README.md

```diff
@@ -1,17 +1,115 @@
 
-The operator tries to be useful out of the box by creating a working default deployment based on the cluster's configuration.
+To provide this functionality, Ingress Operator deploys and manages an
+[OpenShift router](https://github.com/openshift/router) — a
+[HAProxy-based](https://www.haproxy.com) Kubernetes [ingress
+controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers).
 
-* On cloud platforms, ingress is exposed externally using Kubernetes [LoadBalancer Services](https://kubernetes.io/docs/concepts/services-networking/#loadbalancer).
-* On Amazon AWS, the operator publishes a wildcard DNS record based on the [cluster ingress domain](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go) and pointing to the service's load-balancer.
+Ingress Operator implements the OpenShift [ingresscontroller API](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go).
 
-## How to help
+## Installing
 
-See [HACKING.md](HACKING.md) for development topics.
+Ingress Operator is a core feature of OpenShift and is enabled out of the box.
+
+Every new [OpenShift installation](https://github.com/openshift/installer)
+ has an `ingresscontroller` named `default` which can be customized,
+replaced, or supplemented with additional ingress controllers. To view the
+default ingress controller, use the `oc` command:
+
+```shell
+$ oc describe --namespace=openshift-ingress-operator ingresscontroller/default
+```
+
+## Managing
+
+Create and edit `ingresscontroller.operator.openshift.io` resources to manage
+ingress controllers.
+
+Interact with ingress controllers using the `oc` command. Every ingress
+controller lives in the `openshift-ingress-operator` namespace.
+
+To scale an ingress controller:
+
+```shell
+$ oc scale --replicas=1 --namespace=openshift-ingress-operator ingresscontroller/<name>
+```
+
+## Customizing
+
+Create new `ingresscontroller` resources in the `openshift-ingress-operator`
+namespace.
+
+To edit an existing ingress controller:
+
+```shell
+$ oc edit --namespace=openshift-ingress-operator ingresscontroller/<name>
+```
+
+**Important:** Updating an ingress controller may lead to disruption for public
+facing network connections as a new ingress controller revision may be rolled
+out.
+
+Refer to the [ingresscontroller API](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go) for full details on defaults and
+customizing an ingress controller. The most important initial customizations are
+domain and endpoint publishing strategy, as they *cannot currently be changed
+after the ingress controller is created*.
+
+### Endpoint publishing
+
+The `.spec.endpointPublishingStrategy` field is used to publish the ingress
```

Comment:
I'm not sure what "publish the ingress controller endpoints to other networks" means; is that referring to DNS?

Reply:
Basically lifted from our new [public API](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L76) docs; do you feel the API docs are similarly confusing? What would you propose here as an alternative, and would you suggest we upstream a change as well?